s3_scrubber/src/checks.rs

```diff
@@ -307,7 +304,9 @@ pub(crate) async fn list_timeline_blobs(
         errors.push("S3 list response got no index_part.json file".to_string());
     }
 
-    if let Some(index_part_object_key) = index_part_object.as_ref().and_then(|object| object.key())
+    if let Some(index_part_object_key) = index_part_object
+        .as_ref()
+        .and_then(|object| Some(object.key()))
```

Comment:
fixed it

Reply:
Was this a test failure? Your comment predates the test report.

Reply:
Ok perhaps this was just an API change?

Reply:
Later: [aab4a34](https://github.com/neondatabase/neon/pull/5976/commits/aab4a346136f16c59d8523c29c43c132d1083907)